modules/dsl-api/src/main/java/org/mule/runtime/dsl/api/component/ComponentBuildingDefinition.java

```diff
@@ -58,6 +59,7 @@
   private Optional<TypeConverter> typeConverter = empty();
   private Optional<TypeConverter> keyTypeConverter = empty();
   private boolean alwaysEnabled = false;
+  private String gloablName;
```

Comment:
I would call it registration name instead. And I think that we should comment in the javadoc that is expected to allow a single element created from this definition in the whole artifact.